content/posts/scientific-python/will_tirone/index.md

```diff
@@ -0,0 +1,17 @@
+---
+title: "Your Code Could Go To Space"
+date: 2023-07-19
+draft: false
+tags: ["Open-source", "Scientific-Python", "NASA", "JPL"]
+description: "How I learned to code and contributed to a library used on the Mars 2020 mission."
+displayInList: true
+author: ["Will Tirone"]
+---
+
+In mid-2018 I started learning Python by reading textbooks and watching online tutorials. I had absolutely zero background in computer science, but it seemed interesting so I continued to try. At some point, I decided I wanted to do a master’s degree in statistics, so I began to work on more statistics-based programming. That’s when I found SciPy. I became (and still am) fascinated by the idea of open-source software that is completely free to use and supported by a community of diligent programmers. With plenty of extra time on my hands during the pandemic, I made it my goal to contribute to a Python library. My first contribution was actually to a project called [first contributions](https://github.com/firstcontributions/first-contributions) which walks you through a very basic commit and push to GitHub. That built up my confidence a bit, so I decided to tackle a SciPy issue.
+It was not easy. I watched several videos and guides on how to contribute to an open-source library but got stuck many times along the way! I have to admit I felt incompetent trying to make changes to this huge library, but the maintainers and community could not have been nicer or more supportive. That’s really the magic of open source. I was confused and lost, but the (largely volunteer) community was amazing.
+
+Eventually, I managed to get a very small commit merged into the main branch of SciPy. Despite being, at most, a few lines of code, this was a huge landmark for me as a programmer. To my surprise though, in early 2021, a little badge pop up on my GitHub profile that said “Mars 2020 Helicopter Contributor”. I was confused. I didn’t recall working on helicopters, much less helicopters that flew on Mars. I still remember getting chills when I read that I contributed to a library that was used on NASA’s Mars 2020 mission, which involved the robotic helicopter Ingenuity. GitHub posted an [article](https://github.blog/2021-04-19-open-source-goes-to-mars/) explaining how about 12,000 people received a badge indicating that they had contributed to an open-source library that was used on the mission! Keep in mind that I made an absolutely tiny contribution, but I was extremely proud to be recognized in that way.
```

Comment:
Is this your first PR? https://github.com/scipy/scipy/pull/12962

If you are ok linking to it, I think it would be super useful. But no pressure.

Reply:
Or maybe https://github.com/scipy/scipy/pull/12724

Reply:
ah yes good suggestion, I hadn't thought of that.